src/playlist-loader.js

```diff
@@ -104,6 +105,13 @@
 
       // update the playlist loader's state in response to a new or
       // updated playlist.
+      haveMetadataFirst = function(error, xhr, url) {
+        if (error) {
+          loader.state = 'HAVE_MASTER';
```

Comment:
Temporarily I changed the code back to what you had originally. Lets talk about how you want this code to be improved because I am not quite sure I understand exactly what needs to be improved.